### ISSUE TITLE
[1.16] Embeddium crash with "any" mod  

### ISSUE THREAD

**Issue by LanceLuckiest** [FiniteReality/embeddium#206]
### Bug Description

I updated to version 0.3.2+mc1.16.5 of embeddium and after that the game cannot start in any way, it crashes at the time of loading. I have changed and tested all the mods in my modpack one by one and it "fixes" when I remove atmospheric from abnormals but when I try to test only embeddium and atmospheric it works and in the same way with many mods I was testing and always getting the same result, and I couldn't reach any conclusion.

### Reproduction Steps

unknown

### Log File

[crash-2024-02-12_21.19.12-client.txt](https://github.com/embeddedt/embeddium/files/14255117/crash-2024-02-12_21.19.12-client.txt)
[
[latest.log](https://github.com/embeddedt/embeddium/files/14255140/latest.log)
](url)

**Comment by embeddedt** [comment]
Send the full `latest.log`, please.

**Comment by LanceLuckiest** [comment]
done

**Comment by embeddedt** [comment]
Weird, I looked at the log and don't really see how Embeddium 0.3.2 would cause this (there is no earlier error that would be likely to trigger it). Not sure what to suggest.

**Comment by LanceLuckiest** [comment]
I think the same, I just downgraded the mod to 0.3.0 and now it works

**Comment by LanceLuckiest** [comment]
The problem is very strange, I have been dividing the list of mods in my modpack and nothing always has the same result, without embeddium or in its version 0.3.0 this works. btw using curseforge as launcher

[latest.log](https://github.com/embeddedt/embeddium/files/14255611/latest.log)

[crash-2024-02-12_22.16.04-client.txt](https://github.com/embeddedt/embeddium/files/14255615/crash-2024-02-12_22.16.04-client.txt)

**Comment by dos54** [comment]
Updated from 0.3.1 to 0.3.2 for minecraft 1.20.1, game instantly crashed before even getting to the load screen. Downgrading back to 0.3.1 fixed the issue. Last lines in the log file are: 
```
[18:33:22] [main/ERROR]: Error encountered during mixin config postInit step'embeddium.mixins.json': Cannot invoke "net.minecraftforge.fml.loading.moddiscovery.ModFileInfo.getFile()" because the return value of "net.minecraftforge.fml.loading.LoadingModList.getModFileById(String)" is null
java.lang.NullPointerException: Cannot invoke "net.minecraftforge.fml.loading.moddiscovery.ModFileInfo.getFile()" because the return value of "net.minecraftforge.fml.loading.LoadingModList.getModFileById(String)" is null
	at me.jellysquid.mods.sodium.mixin.SodiumMixinPlugin.getMixins(SodiumMixinPlugin.java:108) ~[embeddium-0.3.2+mc1.20.1.jar%23443!/:?]
	at org.spongepowered.asm.mixin.transformer.PluginHandle.getMixins(PluginHandle.java:128) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.postInitialise(MixinConfig.java:794) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.prepareConfigs(MixinProcessor.java:567) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.select(MixinProcessor.java:462) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.checkSelect(MixinProcessor.java:438) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:290) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:250) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.service.modlauncher.MixinTransformationHandler.processClass(MixinTransformationHandler.java:131) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.launch.MixinLaunchPluginLegacy.processClass(MixinLaunchPluginLegacy.java:131) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at cpw.mods.modlauncher.serviceapi.ILaunchPluginService.processClassWithFlags(ILaunchPluginService.java:156) ~[modlauncher-10.0.9.jar:10.0.9+10.0.9+main.dcd20f30]
	at cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:88) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.TransformingClassLoader.maybeTransformClassBytes(TransformingClassLoader.java:50) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.cl.ModuleClassLoader.readerToClass(ModuleClassLoader.java:113) ~[securejarhandler-2.1.10.jar:?]
	at cpw.mods.cl.ModuleClassLoader.lambda$findClass$15(ModuleClassLoader.java:219) ~[securejarhandler-2.1.10.jar:?]
	at cpw.mods.cl.ModuleClassLoader.loadFromModule(ModuleClassLoader.java:229) ~[securejarhandler-2.1.10.jar:?]
	at cpw.mods.cl.ModuleClassLoader.findClass(ModuleClassLoader.java:219) ~[securejarhandler-2.1.10.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:637) ~[?:?]
	at java.lang.Class.forName(Class.java:545) ~[?:?]
	at net.minecraftforge.fml.earlydisplay.DisplayWindow.lambda$updateModuleReads$18(DisplayWindow.java:586) ~[fmlearlydisplay-1.20.1-47.2.0.jar:1.0]
	at java.util.Optional.map(Optional.java:260) ~[?:?]
	at net.minecraftforge.fml.earlydisplay.DisplayWindow.updateModuleReads(DisplayWindow.java:586) ~[fmlearlydisplay-1.20.1-47.2.0.jar:1.0]
	at net.minecraftforge.fml.loading.ImmediateWindowHandler.acceptGameLayer(ImmediateWindowHandler.java:71) ~[fmlloader-1.20.1-47.2.0.jar:1.0]
	at net.minecraftforge.fml.loading.FMLLoader.beforeStart(FMLLoader.java:207) ~[fmlloader-1.20.1-47.2.0.jar:1.0]
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.launchService(CommonLaunchHandler.java:92) ~[fmlloader-1.20.1-47.2.0.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?]
[18:33:22] [main/INFO]: Initializing MixinExtras via com.llamalad7.mixinextras.service.MixinExtrasServiceImpl(version=0.3.2).
[18:33:24] [main/WARN]: @Inject(@At("INVOKE_ASSIGN")) Shift.BY=3 on jeed-common.mixins.json:EffectRenderingInventoryScreenMixin::handler$bie000$captureEffect exceeds the maximum allowed value: 0. Increase the value of maxShiftBy to suppress this warning.
[18:33:26] [Datafixer Bootstrap/INFO]: 188 Datafixer optimizations took 81 milliseconds

```
Full log is included below
[2024-02-12-6.log.gz](https://github.com/embeddedt/embeddium/files/14255848/2024-02-12-6.log.gz)

**Comment by Nolij** [comment]
> The problem is very strange, I have been dividing the list of mods in my modpack and nothing always has the same result, without embeddium or in its version 0.3.0 this works. btw using curseforge as launcher
> 
> [latest.log](https://github.com/embeddedt/embeddium/files/14255611/latest.log)
> 
> [crash-2024-02-12_22.16.04-client.txt](https://github.com/embeddedt/embeddium/files/14255615/crash-2024-02-12_22.16.04-client.txt)

Can you try downgrading to Forge `36.2.39`? `36.2.41` has been known to cause some problems.

**Comment by Nolij** [comment]
> Updated from 0.3.1 to 0.3.2 for minecraft 1.20.1, game instantly crashed before even getting to the load screen. Downgrading back to 0.3.1 fixed the issue. Last lines in the log file are:
> 
> ```
> [18:33:22] [main/ERROR]: Error encountered during mixin config postInit step'embeddium.mixins.json': Cannot invoke "net.minecraftforge.fml.loading.moddiscovery.ModFileInfo.getFile()" because the return value of "net.minecraftforge.fml.loading.LoadingModList.getModFileById(String)" is null
> java.lang.NullPointerException: Cannot invoke "net.minecraftforge.fml.loading.moddiscovery.ModFileInfo.getFile()" because the return value of "net.minecraftforge.fml.loading.LoadingModList.getModFileById(String)" is null
> 	at me.jellysquid.mods.sodium.mixin.SodiumMixinPlugin.getMixins(SodiumMixinPlugin.java:108) ~[embeddium-0.3.2+mc1.20.1.jar%23443!/:?]
> 	at org.spongepowered.asm.mixin.transformer.PluginHandle.getMixins(PluginHandle.java:128) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
> 	at org.spongepowered.asm.mixin.transformer.MixinConfig.postInitialise(MixinConfig.java:794) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
> 	at org.spongepowered.asm.mixin.transformer.MixinProcessor.prepareConfigs(MixinProcessor.java:567) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
> 	at org.spongepowered.asm.mixin.transformer.MixinProcessor.select(MixinProcessor.java:462) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
> 	at org.spongepowered.asm.mixin.transformer.MixinProcessor.checkSelect(MixinProcessor.java:438) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
> 	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:290) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
> 	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:250) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
> 	at org.spongepowered.asm.service.modlauncher.MixinTransformationHandler.processClass(MixinTransformationHandler.java:131) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
> 	at org.spongepowered.asm.launch.MixinLaunchPluginLegacy.processClass(MixinLaunchPluginLegacy.java:131) ~[mixin-0.8.5.jar:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
> 	at cpw.mods.modlauncher.serviceapi.ILaunchPluginService.processClassWithFlags(ILaunchPluginService.java:156) ~[modlauncher-10.0.9.jar:10.0.9+10.0.9+main.dcd20f30]
> 	at cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:88) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.TransformingClassLoader.maybeTransformClassBytes(TransformingClassLoader.java:50) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.cl.ModuleClassLoader.readerToClass(ModuleClassLoader.java:113) ~[securejarhandler-2.1.10.jar:?]
> 	at cpw.mods.cl.ModuleClassLoader.lambda$findClass$15(ModuleClassLoader.java:219) ~[securejarhandler-2.1.10.jar:?]
> 	at cpw.mods.cl.ModuleClassLoader.loadFromModule(ModuleClassLoader.java:229) ~[securejarhandler-2.1.10.jar:?]
> 	at cpw.mods.cl.ModuleClassLoader.findClass(ModuleClassLoader.java:219) ~[securejarhandler-2.1.10.jar:?]
> 	at java.lang.ClassLoader.loadClass(ClassLoader.java:637) ~[?:?]
> 	at java.lang.Class.forName(Class.java:545) ~[?:?]
> 	at net.minecraftforge.fml.earlydisplay.DisplayWindow.lambda$updateModuleReads$18(DisplayWindow.java:586) ~[fmlearlydisplay-1.20.1-47.2.0.jar:1.0]
> 	at java.util.Optional.map(Optional.java:260) ~[?:?]
> 	at net.minecraftforge.fml.earlydisplay.DisplayWindow.updateModuleReads(DisplayWindow.java:586) ~[fmlearlydisplay-1.20.1-47.2.0.jar:1.0]
> 	at net.minecraftforge.fml.loading.ImmediateWindowHandler.acceptGameLayer(ImmediateWindowHandler.java:71) ~[fmlloader-1.20.1-47.2.0.jar:1.0]
> 	at net.minecraftforge.fml.loading.FMLLoader.beforeStart(FMLLoader.java:207) ~[fmlloader-1.20.1-47.2.0.jar:1.0]
> 	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.launchService(CommonLaunchHandler.java:92) ~[fmlloader-1.20.1-47.2.0.jar:?]
> 	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?]
> [18:33:22] [main/INFO]: Initializing MixinExtras via com.llamalad7.mixinextras.service.MixinExtrasServiceImpl(version=0.3.2).
> [18:33:24] [main/WARN]: @Inject(@At("INVOKE_ASSIGN")) Shift.BY=3 on jeed-common.mixins.json:EffectRenderingInventoryScreenMixin::handler$bie000$captureEffect exceeds the maximum allowed value: 0. Increase the value of maxShiftBy to suppress this warning.
> [18:33:26] [Datafixer Bootstrap/INFO]: 188 Datafixer optimizations took 81 milliseconds
> ```
> 
> Full log is included below [2024-02-12-6.log.gz](https://github.com/embeddedt/embeddium/files/14255848/2024-02-12-6.log.gz)

This appears to be a different issue. Please open a new issue, as a fix to this one will likely not solve the crash you're experiencing.

**Comment by Nolij** [comment]
> > The problem is very strange, I have been dividing the list of mods in my modpack and nothing always has the same result, without embeddium or in its version 0.3.0 this works. btw using curseforge as launcher
> > [latest.log](https://github.com/embeddedt/embeddium/files/14255611/latest.log)
> > [crash-2024-02-12_22.16.04-client.txt](https://github.com/embeddedt/embeddium/files/14255615/crash-2024-02-12_22.16.04-client.txt)
> 
> Can you try downgrading to Forge `36.2.39`? `36.2.41` has been known to cause some problems.

I've managed to reproduce the issue locally. Try disabling Spartan Shields and see if that fixes the crash.

**Comment by embeddedt** [comment]
@dos54 Your issue is caused by having Oculus 1.6.15 installed, Embeddium requires a newer version than that.

```
[18:33:18] [main/ERROR]: Unsupported installed optional dependencies:
	Mod ID: 'oculus', Requested by: 'embeddium', Expected range: '(1.6.15,)', Actual version: '1.6.15'
```

However, the mod should indeed not crash in this case, and instead display a proper error screen. I'll fix this in 0.3.3 shortly.

**Comment by LanceLuckiest** [comment]
> > > The problem is very strange, I have been dividing the list of mods in my modpack and nothing always has the same result, without embeddium or in its version 0.3.0 this works. btw using curseforge as launcher
> > > [latest.log](https://github.com/embeddedt/embeddium/files/14255611/latest.log)
> > > [crash-2024-02-12_22.16.04-client.txt](https://github.com/embeddedt/embeddium/files/14255615/crash-2024-02-12_22.16.04-client.txt)
> > 
> > 
> > Can you try downgrading to Forge `36.2.39`? `36.2.41` has been known to cause some problems.
> 
> I've managed to reproduce the issue locally. Try disabling Spartan Shields and see if that fixes the crash.

alright

**Comment by LanceLuckiest** [comment]
> > > The problem is very strange, I have been dividing the list of mods in my modpack and nothing always has the same result, without embeddium or in its version 0.3.0 this works. btw using curseforge as launcher
> > > [latest.log](https://github.com/embeddedt/embeddium/files/14255611/latest.log)
> > > [crash-2024-02-12_22.16.04-client.txt](https://github.com/embeddedt/embeddium/files/14255615/crash-2024-02-12_22.16.04-client.txt)
> > 
> > 
> > Can you try downgrading to Forge `36.2.39`? `36.2.41` has been known to cause some problems.
> 
> I've managed to reproduce the issue locally. Try disabling Spartan Shields and see if that fixes the crash.

nothing, same result

**Comment by Nolij** [comment]
> > > > The problem is very strange, I have been dividing the list of mods in my modpack and nothing always has the same result, without embeddium or in its version 0.3.0 this works. btw using curseforge as launcher
> > > > [latest.log](https://github.com/embeddedt/embeddium/files/14255611/latest.log)
> > > > [crash-2024-02-12_22.16.04-client.txt](https://github.com/embeddedt/embeddium/files/14255615/crash-2024-02-12_22.16.04-client.txt)
> > > 
> > > 
> > > Can you try downgrading to Forge `36.2.39`? `36.2.41` has been known to cause some problems.
> > 
> > 
> > I've managed to reproduce the issue locally. Try disabling Spartan Shields and see if that fixes the crash.
> 
> nothing, same result

Thanks for confirming that. We believe we've found the cause. We're testing a fix.

**Comment by embeddedt** [comment]
This should be fixed in 0.3.3. Tested with the given set of mods (thanks @Nolij).
